### PR TITLE
109 Increase distance when car stops

### LINKED ISSUE #108: Parking algorithm ends prematurely
The parking algorithm ends after the car straightens out rather than when the car is successfullly iniside the parking sapce.

### PR DESCRIPTION
**Pull request by Druner**
Added a "while (!isObsAtFront(35))" that should cause the car to stop when front.getDistance() <= 35 instead of 15. This while loop also fixes #108 as the car no longer relies on checkObstacles() from the main loop() as it uses its own obstacle detection from the front.

This closes issue #108 #109 

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change

### CODE REVIEW

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -437,8 +436,14 @@ void autoRightPark(){ // the car is supposed to park inside a parking spot to it
         frontTimer++;
         distanceTraveled = newDistanceTraveled;
     }
-    car.setAngle(0);
-    car.setSpeed(10);
+    turningAngle = 0;
+    car.setAngle(turningAngle);
+    car.setSpeed(PARKING_SPEED);
+    while (!isObsAtFront(20)){
```
Comment (sh4r10) [inline arduino/smartcar/smartcar.ino:442]: remove curly brakcets, and perhaps add a comment to further elaborate as to what this is doing

File: arduino/smartcar/smartcar.ino [hunk 2]
```diff
@@ -513,14 +518,17 @@ void autoLeftPark(){ // the car is supposed to park inside a parking spot to its
...
         distanceTraveled = newDistanceTraveled;
     }
-    Serial.println("done!");
-    car.setAngle(0);
-    car.setSpeed(PARKING_SPEED);
     // here the car will have the correct angle, car will drive foward and park
+    turningAngle = 0;
+    car.setAngle(turningAngle);
+    car.setSpeed(PARKING_SPEED);
+    while (!isObsAtFront(35)){
```
Comment (sh4r10) [inline arduino/smartcar/smartcar.ino:525]: This should be the same as the the while loop above at line 422.

**Review by sh4r10 — CHANGES_REQUESTED** [review]

File: arduino/smartcar/smartcar.ino (hunk 2 above)
Comment (Druner) [reply arduino/smartcar/smartcar.ino]: Fixed with https://github.com/DIT113-V22/group-02/pull/111/commits/590e4f481dd88a18fc2cef463aa02616c63bbbbb

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (Druner) [reply arduino/smartcar/smartcar.ino]: Fixed with https://github.com/DIT113-V22/group-02/pull/111/commits/590e4f481dd88a18fc2cef463aa02616c63bbbbb

**Review by sh4r10 — APPROVED** [review]

**Pull request merged**